cli/src/validation.ts

```diff
@@ -0,0 +1,311 @@
+    assertChain(fromChain);
+
+    let missing: MissingImplicitConfig = {
+      managerPeers: [],
+      transceiverPeers: [],
+      solanaWormholeTransceiver: false,
+      solanaUpdateLUT: false,
+    };
+
+    if (chainToPlatform(fromChain) === "Solana") {
+      const solanaNtt = from.ntt as SolanaNtt<Network, SolanaChains>;
+      const selfWormholeTransceiver = solanaNtt.pdas
+        .registeredTransceiver(new PublicKey(solanaNtt.contracts.ntt!.manager))
+        .toBase58();
+      const registeredSelfTransceiver = await retryWithExponentialBackoff(
+        () =>
+          solanaNtt.connection.getAccountInfo(
+            new PublicKey(selfWormholeTransceiver)
+          ),
+        5,
+        5000
+      );
+      if (registeredSelfTransceiver === null) {
+        count++;
+        missing.solanaWormholeTransceiver = true;
+      }
+
+      const updateLUT = solanaNtt.initializeOrUpdateLUT({
+        payer: new PublicKey(0),
+        owner: new PublicKey(0),
+      });
+      if (!(await updateLUT.next()).done) {
+        count++;
+        missing.solanaUpdateLUT = true;
+      }
+    }
+
+    for (const [toChain, to] of Object.entries(deps)) {
+      if (!to) {
+        continue;
+      }
+      assertChain(toChain);
+      if (fromChain === toChain) {
+        continue;
+      }
+      if (verbose) {
+        process.stdout.write(
+          `Verifying registration for ${fromChain} -> ${toChain}......\n`
+        );
+      }
+      const peer = await retryWithExponentialBackoff(
+        () => from.ntt.getPeer(toChain),
+        5,
+        5000
+      );
+      if (peer === null) {
+        const configLimit = from.config.local?.limits?.inbound?.[
+          toChain
+        ]?.replace(".", "");
+        count++;
+        missing.managerPeers.push({
+          address: to.manager,
+          tokenDecimals: to.decimals,
+          inboundLimit: BigInt(configLimit ?? 0),
+        });
+      } else {
+        if (
+          // @ts-ignore TODO
+          !Buffer.from(peer.address.address.address).equals(
+            // @ts-ignore TODO
+            Buffer.from(to.manager.address.address)
+          )
+        ) {
+          console.error(`Peer address mismatch for ${fromChain} -> ${toChain}`);
+        }
+        if (peer.tokenDecimals !== to.decimals) {
+          console.error(
+            `Peer decimals mismatch for ${fromChain} -> ${toChain}`
+          );
+        }
+      }
+
+      const transceiverPeer = await retryWithExponentialBackoff(
+        () => from.whTransceiver.getPeer(toChain),
+        5,
+        5000
+      );
+      const transceiverAddress = await to.whTransceiver.getAddress();
+      if (transceiverPeer === null) {
+        count++;
+        missing.transceiverPeers.push(transceiverAddress);
+      } else {
+        if (
+          // @ts-ignore TODO
+          !Buffer.from(transceiverPeer.address.address).equals(
+            // @ts-ignore TODO
+            Buffer.from(transceiverAddress.address.address)
```

Comment:
The `// @ts-ignore TODO`s can prolly be removed by adding a converting the address to a string. Something like so:
```suggestion
          !Buffer.from(transceiverPeer.address.address.toString()).equals(
            Buffer.from(transceiverAddress.address.address.toString())
```